gcovr/tests/add_coverages/Makefile

```diff
@@ -0,0 +1,40 @@
+CFLAGS= -fprofile-arcs -ftest-coverage -fPIC
+
+all:
+	$(CXX) $(CFLAGS) -c foo.cpp -o foo.o
+	$(CXX) $(CFLAGS) -c bar.cpp -o bar.o
+	$(CXX) $(CFLAGS) -DFOO -c main.cpp -o main_foo.o
+	$(CXX) $(CFLAGS) main_foo.o foo.o -o testcase_foo
+	$(CXX) $(CFLAGS) -DBAR -c main.cpp -o main_bar.o
+	$(CXX) $(CFLAGS) main_bar.o bar.o -o testcase_bar
+
+run: txt xml html sonarqube json
+
+json_foo:
+	./testcase_foo
+	$(GCOVR) -d --json-pretty -o coverage_foo.json
+
+json_bar:
+	./testcase_bar
+	$(GCOVR) -d --json-pretty -o coverage_bar.json
+
+txt: json_foo json_bar
+	$(GCOVR) -a coverage_foo.json -a coverage_bar.json -o coverage.txt
```

Comment:
I really love this test! It's a great demonstration of the --add-tracefile functionality.

In the future, I hope to port the reference data of the other tests to JSON so that I don't have to check the HTML and XML data every time. This makes the following things possible or a lot easier:

* faster test suite
* smaller test suite
* changes to the HTML format will become easier because fewer files will have to be updated
* supporting multiple compilers in the test suite instead of just GCC 5
* the XML tests will become less important, so I can probably remove the pyutilib dependency \o/

So thank you a lot for working on this cornerstone feature!